File: Java/jakarta/filesystemstorage/src/main/java/com/ithit/webdav/samples/fsstorageservlet/websocket/NotificationEncoder.java
package com.ithit.webdav.samples.fsstorageservlet.websocket;

import jakarta.websocket.EncodeException;
import jakarta.websocket.Encoder;
import jakarta.websocket.EndpointConfig;

/**
 * Encodes notification object to the JSON
 */
public class NotificationEncoder implements Encoder.Text<WebSocketServer.Notification> {

    @Override
    public void init(EndpointConfig config) {
    }

    @Override
    public void destroy() {
    }

    @Override
    public String encode(WebSocketServer.Notification notification) throws EncodeException {
        String target = "";
        if (notification instanceof WebSocketServer.MovedNotification) {
            target = "\"TargetPath\" : \"" + ((WebSocketServer.MovedNotification) notification).getTargetPath() + "\" ,";
        }
        return "{" + target +
                "\"ItemPath\" : \"" + notification.getItemPath() + "\" ," +
                "\"EventType\" : \"" + notification.getOperation() + "\"" +
                "}";
    }
}

File: Kotlin/javax/filesystemstorage/src/main/kotlin/com/ithit/webdav/samples/fsstorageservlet/websocket/NotificationEncoder.kt
package com.ithit.webdav.samples.fsstorageservlet.websocket

import javax.websocket.EncodeException
import javax.websocket.Encoder
import javax.websocket.EndpointConfig

/**
 * Encodes notification object to the JSON
 */
class NotificationEncoder : Encoder.Text<WebSocketServer.Notification> {

    override fun init(config: EndpointConfig) {}

    override fun destroy() {}

    @Throws(EncodeException::class)
    override fun encode(notification: WebSocketServer.Notification): String {
        var target = ""
        if (notification is WebSocketServer.MovedNotification) {
            target = "\"TargetPath\" : \"" + notification.targetPath + "\" ,"
        }
        return "{" + target +
                "\"ItemPath\" : \"" + notification.itemPath + "\" ," +
                "\"EventType\" : \"" + notification.operation + "\"" +
                "}"
    }
}
